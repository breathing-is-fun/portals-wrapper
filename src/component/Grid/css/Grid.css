.Grid iframe {
    border: none;
    width: 100%;
    overflow: hidden;
}

.Grid .render-div {
    overflow: auto;
    padding: 15px;
}