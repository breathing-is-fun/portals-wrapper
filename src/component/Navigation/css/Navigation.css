.Navigation .content {
  height: 65px;
  background: #2e7df3;
  box-shadow: 0px 0px 9px 0px rgba(39, 93, 214, 0.39);
}

.Navigation .content-sm {
  height: 40px;
  background: #2e7df3;
  box-shadow: 0px 0px 9px 0px rgba(39, 93, 214, 0.39);
}

.Navigation .children-wrapper {
  overflow-x: hidden;
  overflow-y: auto;
  background-color: #ffffff;
  box-shadow: 0px 0px 29px 0px rgba(93, 106, 113, 0.12);
  border-radius: 2px;
}

.Navigation .content-wrapper {
  font-size: 28px;
  color: #ffffff;
  float: left;
  height: 100%;
  line-height: 65px;
  text-align: center;
}

.Navigation .sm {
  font-size: 20px;
  color: #ffffff;
  float: left;
  height: 100%;
  line-height: 40px;
  text-align: center;
}

.Navigation .content-switch {
  float: right;
  margin-right: 20px;
  color: #FFF;
}

.Navigation .content-space {
  width: 1px;
  height: 100%;
  background: #2372e7;
  float: left;
}

.Navigation .content-time {
  font-size: 34px;
  color: #ffffff;
}

.Navigation .content-time-detail {
  height: 24px;
  font-size: 13px;
  line-height: 33px;
  color: #ffffff;
}

.Navigation .content-wrapper .droplink {
  color: #FFF;
  background-color: transparent;
  text-decoration: none;
  outline: none;
  cursor: pointer;
}

.Navigation .content-wrapper .droplink .droplink-icon {
  font-size: 18px;
  transition: all 0.3s cubic-bezier(0.645, 0.045, 0.355, 1);
}