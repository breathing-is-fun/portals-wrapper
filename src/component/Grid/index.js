import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { Responsive, WidthProvider } from 'react-grid-layout';
const ResponsiveGridLayout = WidthProvider(Responsive);

import 'react-grid-layout/css/styles.css';
import 'react-resizable/css/styles.css';

import Shell from '../../component/Shell';
import Loader from './ModulesLoader';

import './css/Grid.css';

export default class Grid extends Component {
	constructor (props) {
		super(props);

		this.state = {
			isDrawerOpen: false,
			propertyBoardDataSource: {},
			PropertyBoard: null,
			rowHeight: 30,
		};

		this.roots = {};
	}

	static defaultProps = {
		showEdit: false,
		showDelete: false,
		layout: [],
		enumDatas: [],
	}

	mountRoots = () => {
		const { layout } = this.props;

    	const loader = new Loader(layout, this.roots);

    	loader.load();
	}

	handleLayoutChange = layout => {
		const { onLayoutChange, layout: propsLayout } = this.props;

		let newLayout = [];

		for(let i = 0; i < layout.length; i++) {
			const item = layout[i];
			const jtem = propsLayout[i];

			newLayout.push(Object.assign({}, jtem, item));
		}

		onLayoutChange && onLayoutChange(newLayout);
	}

	handleDragDrop = e => {
		let { layout, onLayoutChange } = this.props;
		const item = JSON.parse(e.dataTransfer.getData('menuItemToGrid'));
		const { key, imgurl: imgUrl, text, style, url } = item;
		const defaultProps = {
			i: '' + key + layout.length,
			x: (layout.length * 2) % 12,
			w: 2, h: 9, y: 0,
			url,
			path: url,
			imgUrl,
			title: text,
			style,
		};

		layout.push(Object.assign({}, item, defaultProps));

		onLayoutChange && onLayoutChange(layout);
	}

	createShellChild = (showEdit, item) => {
		const { i, moduletype: type, imgurl, imgUrl, path } = item;
		const height = 'calc(100% - 21px)';

		if(showEdit) {
			return (
				<img
					src={ imgurl || imgUrl }
					style={{ width: '100%', height }}
				/>
			);
		}

		if(type == 'iframe') {
			return (
				<iframe
					src={ path }
					style={{ height }}
				></iframe>
			);
		}

		return (
			<div
				className='render-div'
				style={{ height }}
				ref={ ref => ref && (this.roots[i] = ref) }
				id={ i }
			/>
		);
	}

	handleShellonEdit = item => {
		import('../PropertyBoard')
			.then(PropertyBoard => {
				this.setState({
					isDrawerOpen: true,
					propertyBoardDataSource: item,
					PropertyBoard: PropertyBoard.default || PropertyBoard,
				});
			});
	}

	handleShellStyleOnChange = currentShellStyle => {
		const { onLayoutChange, layout } = this.props;
		let newLayout = [];

		for(let item of layout) {
			const { i: key, style = {} } = item;
			const { id } = currentShellStyle;
			let obj = item;

			if(key == id) {
				obj.style = Object.assign(style, currentShellStyle);
				delete obj.style.id;
			}

			newLayout.push(obj);
		}

		onLayoutChange && onLayoutChange(newLayout);
	}

	handleBreakpointChange = (breakpoints, cols) => {
		this.setState({ rowHeight: cols * 2.5 });
	}

    render = () => {
    	const {
    		isDrawerOpen,
    		propertyBoardDataSource,
    		PropertyBoard,
    		rowHeight,
    	} = this.state;
    	const {
    		showEdit,
    		showDelete,
    		layout,
    		onDelete,
    		onDetail,
    		enumDatas
    	} = this.props;

    	const layoutProps = {
    		className: 'layout',
    		cols: { lg: 12, md: 10, sm: 6, xs: 4, xxs: 2 },
    		breakpoints: { lg: 1300, md: 996, sm: 768, xs: 480, xxs: 0 },
    		rowHeight,
    		width: document.body.clientWidth - (showEdit ? 256 : 0),
    		// margin: [10, 10],
    		onLayoutChange: this.handleLayoutChange,
    		isDraggable: showEdit,
    		isResizable: showEdit,
    		// compactType: 'horizontal',
    		style: {
    			background: '#f5f6fa',
    			height: layout.length != 0 ? '100%' : 500
    		},
    		onBreakpointChange: this.handleBreakpointChange,
    	};

    	const propertyBoardProps = {
    		visible: isDrawerOpen,
    		onClose: isDrawerOpen => {
    			this.setState({ isDrawerOpen }, () => {
    				// 关闭抽屉时销毁外壳中的元素，好在再次点击时执行componentDidMount中的方法
    				setTimeout(() => {
    					this.setState({ PropertyBoard: null });
    				}, 300);
    			});
    		},
    		shellStyleDatas: propertyBoardDataSource,
    		enumDatas,
    		onChange: this.handleShellStyleOnChange,
    	};

    	return (
    		<div
    			className='Grid'
    			onDrop={ this.handleDragDrop }
    			onDragOver={ e => e.preventDefault() }
    		>
    			<ResponsiveGridLayout { ...layoutProps }>
    				{
    					layout.map(item => {
    						const {
    							i: key,
    							title,
    							style = {},
    							showtitle: showTitle,
    							showdetail: showDetail,
    							detailpath: detailPath,
    						} = item;

    						const shellProps = {
    							key,
    							title,
    							showEdit,
    							showDelete,
    							onDelete,
    							showTitle,
    							onDetail,
    							detailPath,
    							showDetail,
    							style,
    							'data-grid': item,
    							onEdit: dataGrid => {
    								this.handleShellonEdit(dataGrid);
    							},
    						};

    						return (
    							<Shell { ...shellProps }>
    								{ this.createShellChild(showEdit, item) }
    							</Shell>
    						);
    					})
    				}
    			</ResponsiveGridLayout>

    			{ PropertyBoard && <PropertyBoard { ...propertyBoardProps } /> }
    		</div>
    	);
    }
}

Grid.propTypes = {
	layout: PropTypes.array,
	onLayoutChange: PropTypes.func,
	enumDatas: PropTypes.array,
	showEdit: PropTypes.bool,
	showDelete: PropTypes.bool,
	onDelete: PropTypes.func,
	onDetail: PropTypes.func,
};