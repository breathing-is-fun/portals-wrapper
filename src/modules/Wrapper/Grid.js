/*
 * @Author: zy9@github.com/zy410419243
 * @Date: 2018-09-26 11:25:50
 * @Last Modified by: zy9
 * @Last Modified time: 2018-09-29 15:46:01
 */
import React, { Component } from 'react';

import GridLayout from 'react-grid-layout';
import 'react-grid-layout/css/styles.css';
import 'react-resizable/css/styles.css';

export default class Grid extends Component {
	constructor (props) {
		super(props);

		this.state = {
			layout: [],
		};
	}

    componentDidMount = () => {
    	this.loadLayout(() => {
    		const { layout } = this.state;
    		let pathArr = [];

    		for(let item of layout) {
    			const { path, type } = item;

    			type != 'iframe' && pathArr.push(`import('${ path }')`);
    		}
    		pathArr = `[${ pathArr.toString() }]`;

    		/* eslint-disable no-eval */
    		Promise.all(eval(`${ pathArr }`)).then(modules => {
    			for(let i = 0; i < modules.length; i++) {
    				const { TestModule } = modules[i];
    				const { i: key } = layout[i];
    				const testModule = new TestModule(this[key]);
    				const { _moduleOnMount } = testModule;

    				_moduleOnMount && _moduleOnMount.call(testModule);
    			}
    		});
    	});
    }

	loadLayout = callback => {
		fetch('../../../mock/layoutDatas.json')
			.then(result => result.json())
			.then(result => {
				const { layout } = result;

				this.setState({ layout }, () => callback && callback());
			});
	}

    handleLayoutChange = layout => {
    	console.log(layout);
    }

	handleDragDrop = e => {
		let { layout } = this.state;
		const item = JSON.parse(e.dataTransfer.getData('menuItemToGrid'));
		const { key, url, text } = item;

		layout.push({
			i: '' + key + layout.length,
			x: (layout.length * 2) % 12,
			y: Infinity,
			w: 2,
			h: 9,
			type: 'iframe',
			imgUrl: url,
			title: text
		});

		this.setState({ layout });
	}

    render = () => {
    	const { layout } = this.state;

    	const layoutProps = {
    		className: 'layout',
    		// layout,
    		cols: 12,
    		rowHeight: 30,
    		width: (document.documentElement.clientWidth || document.body.clientWidth) - 256,
    		margin: [10, 10],
    		onLayoutChange: this.handleLayoutChange,
    	};

    	return (
    		<div className='Grid' onDrop={ this.handleDragDrop } onDragOver={ e => e.preventDefault() }>
    			<GridLayout { ...layoutProps }>
    				{
    					layout.map(item => {
    						const { i, type, imgUrl, title } = item;

    						return type == 'iframe' ? (
    							<div key={ i } data-grid={ item } style={{ zIndex: 1 }}>
    								<div style={{ height: 30, background: '#F96' }}>{ title }</div>
    								<img src={ imgUrl } style={{ width: '100%', height: 'calc(100% - 30px)' }} />
    							</div>
    						) : (
    							<div key={ i } data-grid={ item } style={{ background: '#ccc' }} ref={ ref => this[i] = ref } />
    						);
    					})
    				}
    			</GridLayout>
    		</div>
    	);
    }
}