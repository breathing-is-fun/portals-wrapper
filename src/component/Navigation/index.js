/*
 * @Author: zy9@github.com/zy410419243
 * @Date: 2018-10-15 15:47:19
 * @Last Modified by: zy9
 * @Last Modified time: 2018-10-18 16:38:28
 */
import React, { Component } from 'react';

import moment from 'moment';
moment.locale('zh-cn');

import './css/Navigation.css';

export default class Navigation extends Component {
	constructor (props) {
		super(props);

		this.state = {
			time: moment().format('HH:mm:ss'),
		};
	}

    componentDidMount = () => {
    	this.getCurrentTime();
    }

    getCurrentTime = () => {
    	setInterval(() => {
    		this.setState({ time: moment().format('HH:mm:ss') });
    	}, 1000);
    }

    render = () => {
    	const { children } = this.props;
    	const { time } = this.state;

    	// 分割线
    	const space = <div className='content-space' />;

    	return (
    		<div className='Navigation'>
    			<div className='content'>
    				<div className='content-wrapper' style={{ width: '15%' }}>浙江省水利综合门户</div>

    				{/* { space } */}
    				<div className='content-wrapper' style={{ width: '15%' }}>
    					<div className='content-time'>{ time }</div>
    				</div>

    				<div className='content-wrapper' style={{ marginLeft: '-3.5%' }}>
    					<div className='content-time-detail'>{ moment().format('YYYY-MM-DD') }</div>
    					<div className='content-time-detail'>{ moment().format('dddd') }</div>
    				</div>
    			</div>

    			{ children }
    		</div>
    	);
    }
}