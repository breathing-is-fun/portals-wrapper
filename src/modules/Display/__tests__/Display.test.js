import React, { Component } from 'react';
import { render, mount, shallow } from 'enzyme';

import Grid from '../../../component/Grid';
import Navigation from '../../../component/Navigation';
import Loader from '../../../component/Grid/ModulesLoader';

describe('Display', () => {
  const layout = [
    {
      detailpath: 'http://www.baidu.com/',
      h: 12,
      i: 'demo1',
      id: 117,
      imgurl: 'http://47.95.1.229:9003/UploadFile/201808/szy.png',
      maxw: 0,
      mealid: 89,
      minw: 0,
      path: '../../../../thirdModules/demo1/index.js',
      showdetail: true,
      still: false,
      style: {},
      title: 'demo1',
      w: 12,
      x: 0,
      y: 0,
    },
    {
      detailpath: 'http://www.baidu.com/',
      h: 9,
      i: 'demo2',
      id: 118,
      imgurl: 'http://47.95.1.229:9003/UploadFile/201808/dzx.png',
      maxw: 0,
      mealid: 89,
      minw: 0,
      path: '../../../../thirdModules/demo2/index.js',
      showdetail: true,
      still: false,
      style: {},
      title: 'demo2',
      w: 3,
      x: 0,
      y: 12,
    },
  ];

  it('render correctly', () => {
    class Demo extends Component {
      render = () => (
        <Navigation
          datas={[
            {
              department: '4,5,1',
              group: '',
              id: 89,
              imgurl: 'http://47.95.1.229:9007/UploadFile/ddzx.jpg',
              isdelete: false,
              isedit: false,
              title: 'test',
              url: '',
            },
          ]}
          title="test"
        >
          <Grid
            showEdit={false}
            showDelete={false}
            ref={ref => (this.grid = ref)}
            layout={layout}
          />
        </Navigation>
      );
    }

    expect(mount(<Demo />)).toMatchSnapshot();
  });
});
