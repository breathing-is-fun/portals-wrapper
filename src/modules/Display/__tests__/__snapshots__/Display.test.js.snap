// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display render correctly 1`] = `
<Demo>
  <Navigation
    clock={false}
    datas={
      Array [
        Object {
          "department": "4,5,1",
          "group": "",
          "id": 89,
          "imgurl": "http://47.95.1.229:9007/UploadFile/ddzx.jpg",
          "isdelete": false,
          "isedit": false,
          "title": "test",
          "url": "",
        },
      ]
    }
    display={true}
    size=""
    title="test"
  >
    <div
      className="Navigation"
    >
      <div
        className="content"
        style={
          Object {
            "display": "",
          }
        }
      >
        <div
          className="content-wrapper"
          style={
            Object {
              "width": 300,
            }
          }
        >
          test
        </div>
        <div
          className="content-wrapper"
          style={
            Object {
              "width": 300,
            }
          }
        >
          <div
            className="content-time"
          />
        </div>
        <div
          className="content-wrapper content-switch"
        >
          <Dropdown
            mouseEnterDelay={0.15}
            mouseLeaveDelay={0.1}
            onVisibleChange={[Function]}
            overlay={
              <Menu
                className=""
                focusable={false}
                theme="light"
              >
                <MenuItem
                  onClick={[Function]}
                >
                  <a
                    rel="noopener noreferrer"
                  >
                    test
                  </a>
                </MenuItem>
              </Menu>
            }
            placement="bottomLeft"
            trigger={
              Array [
                "hover",
                "click",
              ]
            }
            visible={false}
          >
            <Dropdown
              defaultVisible={false}
              mouseEnterDelay={0.15}
              mouseLeaveDelay={0.1}
              onVisibleChange={[Function]}
              overlay={[Function]}
              overlayClassName=""
              overlayStyle={Object {}}
              placement="bottomLeft"
              prefixCls="ant-dropdown"
              showAction={Array []}
              transitionName="slide-up"
              trigger={
                Array [
                  "hover",
                  "click",
                ]
              }
              visible={false}
            >
              <Trigger
                action={
                  Array [
                    "hover",
                    "click",
                  ]
                }
                afterPopupVisibleChange={[Function]}
                blurDelay={0.15}
                builtinPlacements={
                  Object {
                    "bottomCenter": Object {
                      "offset": Array [
                        0,
                        4,
                      ],
                      "overflow": Object {
                        "adjustX": 1,
                        "adjustY": 1,
                      },
                      "points": Array [
                        "tc",
                        "bc",
                      ],
                      "targetOffset": Array [
                        0,
                        0,
                      ],
                    },
                    "bottomLeft": Object {
                      "offset": Array [
                        0,
                        4,
                      ],
                      "overflow": Object {
                        "adjustX": 1,
                        "adjustY": 1,
                      },
                      "points": Array [
                        "tl",
                        "bl",
                      ],
                      "targetOffset": Array [
                        0,
                        0,
                      ],
                    },
                    "bottomRight": Object {
                      "offset": Array [
                        0,
                        4,
                      ],
                      "overflow": Object {
                        "adjustX": 1,
                        "adjustY": 1,
                      },
                      "points": Array [
                        "tr",
                        "br",
                      ],
                      "targetOffset": Array [
                        0,
                        0,
                      ],
                    },
                    "topCenter": Object {
                      "offset": Array [
                        0,
                        -4,
                      ],
                      "overflow": Object {
                        "adjustX": 1,
                        "adjustY": 1,
                      },
                      "points": Array [
                        "bc",
                        "tc",
                      ],
                      "targetOffset": Array [
                        0,
                        0,
                      ],
                    },
                    "topLeft": Object {
                      "offset": Array [
                        0,
                        -4,
                      ],
                      "overflow": Object {
                        "adjustX": 1,
                        "adjustY": 1,
                      },
                      "points": Array [
                        "bl",
                        "tl",
                      ],
                      "targetOffset": Array [
                        0,
                        0,
                      ],
                    },
                    "topRight": Object {
                      "offset": Array [
                        0,
                        -4,
                      ],
                      "overflow": Object {
                        "adjustX": 1,
                        "adjustY": 1,
                      },
                      "points": Array [
                        "br",
                        "tr",
                      ],
                      "targetOffset": Array [
                        0,
                        0,
                      ],
                    },
                  }
                }
                defaultPopupVisible={false}
                defaultVisible={false}
                destroyPopupOnHide={false}
                focusDelay={0}
                getDocument={[Function]}
                getPopupClassNameFromAlign={[Function]}
                hideAction={Array []}
                mask={false}
                maskClosable={true}
                mouseEnterDelay={0.15}
                mouseLeaveDelay={0.1}
                onPopupAlign={[Function]}
                onPopupVisibleChange={[Function]}
                onVisibleChange={[Function]}
                overlay={[Function]}
                popup={[Function]}
                popupAlign={Object {}}
                popupClassName=""
                popupPlacement="bottomLeft"
                popupStyle={Object {}}
                popupTransitionName="slide-up"
                popupVisible={false}
                prefixCls="ant-dropdown"
                showAction={Array []}
                visible={false}
              >
                <span
                  className="droplink ant-dropdown-trigger"
                  key="trigger"
                  onClick={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onTouchStart={[Function]}
                >
                  test
                  <Icon
                    className="droplink-icon"
                    style={
                      Object {
                        "marginLeft": 10,
                        "transform": "rotate(0deg)",
                      }
                    }
                    type="down"
                  >
                    <i
                      className="anticon anticon-down droplink-icon"
                      style={
                        Object {
                          "marginLeft": 10,
                          "transform": "rotate(0deg)",
                        }
                      }
                    >
                      <IconReact
                        className=""
                        type="down-o"
                      >
                        <svg
                          aria-hidden="true"
                          className=""
                          data-icon="down"
                          fill="currentColor"
                          height="1em"
                          key="svg-down"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                            key="svg-down-svg-0"
                          />
                        </svg>
                      </IconReact>
                    </i>
                  </Icon>
                </span>
              </Trigger>
            </Dropdown>
          </Dropdown>
        </div>
      </div>
      <div
        className="children-wrapper"
        style={
          Object {
            "height": "-65px",
          }
        }
      >
        <Grid
          enumDatas={Array []}
          layout={
            Array [
              Object {
                "detailpath": "http://www.baidu.com/",
                "h": 12,
                "i": "demo1",
                "id": 117,
                "imgurl": "http://47.95.1.229:9003/UploadFile/201808/szy.png",
                "maxw": 0,
                "mealid": 89,
                "minw": 0,
                "path": "../../../../thirdModules/demo1/index.js",
                "showdetail": true,
                "still": false,
                "style": Object {},
                "title": "demo1",
                "w": 12,
                "x": 0,
                "y": 0,
              },
              Object {
                "detailpath": "http://www.baidu.com/",
                "h": 9,
                "i": "demo2",
                "id": 118,
                "imgurl": "http://47.95.1.229:9003/UploadFile/201808/dzx.png",
                "maxw": 0,
                "mealid": 89,
                "minw": 0,
                "path": "../../../../thirdModules/demo2/index.js",
                "showdetail": true,
                "still": false,
                "style": Object {},
                "title": "demo2",
                "w": 3,
                "x": 0,
                "y": 12,
              },
            ]
          }
          showDelete={false}
          showEdit={false}
          size=""
        >
          <div
            className="Grid"
            onDragOver={[Function]}
            onDrop={[Function]}
          >
            <WidthProvider
              breakpoints={
                Object {
                  "lg": 1270,
                  "md": 996,
                  "sm": 768,
                  "xs": 480,
                  "xxs": 0,
                }
              }
              className="layout"
              cols={
                Object {
                  "lg": 12,
                  "md": 10,
                  "sm": 6,
                  "xs": 4,
                  "xxs": 2,
                }
              }
              isDraggable={false}
              isResizable={false}
              margin={
                Array [
                  20,
                  25,
                ]
              }
              measureBeforeMount={false}
              onBreakpointChange={[Function]}
              onLayoutChange={[Function]}
              rowHeight={5}
              style={
                Object {
                  "background": "#f5f6fa",
                  "height": "100%",
                }
              }
              width={0}
            >
              <ResponsiveReactGridLayout
                breakpoints={
                  Object {
                    "lg": 1270,
                    "md": 996,
                    "sm": 768,
                    "xs": 480,
                    "xxs": 0,
                  }
                }
                className="layout"
                cols={
                  Object {
                    "lg": 12,
                    "md": 10,
                    "sm": 6,
                    "xs": 4,
                    "xxs": 2,
                  }
                }
                isDraggable={false}
                isResizable={false}
                layouts={
                  Object {
                    "lg": Array [],
                    "xxs": Array [
                      Object {
                        "h": 12,
                        "i": "demo1",
                        "isDraggable": undefined,
                        "isResizable": undefined,
                        "maxH": undefined,
                        "maxW": undefined,
                        "minH": undefined,
                        "minW": undefined,
                        "moved": false,
                        "static": false,
                        "w": 2,
                        "x": 0,
                        "y": 0,
                      },
                      Object {
                        "h": 9,
                        "i": "demo2",
                        "isDraggable": undefined,
                        "isResizable": undefined,
                        "maxH": undefined,
                        "maxW": undefined,
                        "minH": undefined,
                        "minW": undefined,
                        "moved": false,
                        "static": false,
                        "w": 2,
                        "x": 0,
                        "y": 12,
                      },
                    ],
                  }
                }
                margin={
                  Array [
                    20,
                    25,
                  ]
                }
                onBreakpointChange={[Function]}
                onLayoutChange={[Function]}
                onWidthChange={[Function]}
                rowHeight={5}
                style={
                  Object {
                    "background": "#f5f6fa",
                    "height": "100%",
                  }
                }
                width={0}
              >
                <ReactGridLayout
                  autoSize={true}
                  className="layout"
                  cols={2}
                  compactType="vertical"
                  containerPadding={null}
                  draggableCancel=""
                  draggableHandle=""
                  isDraggable={false}
                  isResizable={false}
                  layout={
                    Array [
                      Object {
                        "h": 12,
                        "i": "demo1",
                        "isDraggable": undefined,
                        "isResizable": undefined,
                        "maxH": undefined,
                        "maxW": undefined,
                        "minH": undefined,
                        "minW": undefined,
                        "moved": false,
                        "static": false,
                        "w": 2,
                        "x": 0,
                        "y": 0,
                      },
                      Object {
                        "h": 9,
                        "i": "demo2",
                        "isDraggable": undefined,
                        "isResizable": undefined,
                        "maxH": undefined,
                        "maxW": undefined,
                        "minH": undefined,
                        "minW": undefined,
                        "moved": false,
                        "static": false,
                        "w": 2,
                        "x": 0,
                        "y": 12,
                      },
                    ]
                  }
                  margin={
                    Array [
                      20,
                      25,
                    ]
                  }
                  maxRows={Infinity}
                  onDrag={[Function]}
                  onDragStart={[Function]}
                  onDragStop={[Function]}
                  onLayoutChange={[Function]}
                  onResize={[Function]}
                  onResizeStart={[Function]}
                  onResizeStop={[Function]}
                  preventCollision={false}
                  rowHeight={5}
                  style={
                    Object {
                      "background": "#f5f6fa",
                      "height": "100%",
                    }
                  }
                  useCSSTransforms={true}
                  verticalCompact={true}
                  width={0}
                >
                  <div
                    className="react-grid-layout layout"
                    style={
                      Object {
                        "background": "#f5f6fa",
                        "height": "100%",
                      }
                    }
                  >
                    <GridItem
                      cancel=""
                      className=""
                      cols={2}
                      containerPadding={
                        Array [
                          20,
                          25,
                        ]
                      }
                      containerWidth={0}
                      h={12}
                      handle=""
                      i="demo1"
                      isDraggable={false}
                      isResizable={false}
                      key=".$demo1"
                      margin={
                        Array [
                          20,
                          25,
                        ]
                      }
                      maxH={Infinity}
                      maxRows={Infinity}
                      maxW={Infinity}
                      minH={1}
                      minW={1}
                      onDrag={[Function]}
                      onDragStart={[Function]}
                      onDragStop={[Function]}
                      onResize={[Function]}
                      onResizeStart={[Function]}
                      onResizeStop={[Function]}
                      rowHeight={5}
                      static={false}
                      useCSSTransforms={true}
                      usePercentages={false}
                      w={2}
                      x={0}
                      y={0}
                    >
                      <Shell
                        className="react-grid-item cssTransforms"
                        data-grid={
                          Object {
                            "detailpath": "http://www.baidu.com/",
                            "h": 12,
                            "i": "demo1",
                            "id": 117,
                            "imgurl": "http://47.95.1.229:9003/UploadFile/201808/szy.png",
                            "maxw": 0,
                            "mealid": 89,
                            "minw": 0,
                            "path": "../../../../thirdModules/demo1/index.js",
                            "showdetail": true,
                            "still": false,
                            "style": Object {},
                            "title": "demo1",
                            "w": 12,
                            "x": 0,
                            "y": 0,
                          }
                        }
                        detailPath="http://www.baidu.com/"
                        key="demo1"
                        onEdit={[Function]}
                        showDelete={false}
                        showDetail={true}
                        showEdit={false}
                        showTitle={true}
                        size=""
                        style={
                          Object {
                            "MozTransform": "translate(20px,25px)",
                            "OTransform": "translate(20px,25px)",
                            "WebkitTransform": "translate(20px,25px)",
                            "height": "335px",
                            "msTransform": "translate(20px,25px)",
                            "position": "absolute",
                            "transform": "translate(20px,25px)",
                            "width": "-40px",
                          }
                        }
                        title="demo1"
                        type="component"
                      >
                        <div
                          className="Shell"
                          data-grid={
                            Object {
                              "detailpath": "http://www.baidu.com/",
                              "h": 12,
                              "i": "demo1",
                              "id": 117,
                              "imgurl": "http://47.95.1.229:9003/UploadFile/201808/szy.png",
                              "maxw": 0,
                              "mealid": 89,
                              "minw": 0,
                              "path": "../../../../thirdModules/demo1/index.js",
                              "showdetail": true,
                              "still": false,
                              "style": Object {},
                              "title": "demo1",
                              "w": 12,
                              "x": 0,
                              "y": 0,
                            }
                          }
                          onClick={[Function]}
                          size=""
                          style={
                            Object {
                              "MozTransform": "translate(20px,25px)",
                              "OTransform": "translate(20px,25px)",
                              "WebkitTransform": "translate(20px,25px)",
                              "height": "335px",
                              "msTransform": "translate(20px,25px)",
                              "overflow": "auto",
                              "position": "absolute",
                              "transform": "translate(20px,25px)",
                              "width": "-40px",
                            }
                          }
                          type="component"
                        >
                          <span
                            onMouseDown={[Function]}
                            style={
                              Object {
                                "cursor": "pointer",
                                "fontSize": 17,
                                "position": "absolute",
                                "right": 10,
                                "top": 6,
                              }
                            }
                          >
                            详情 &gt;
                          </span>
                          <span
                            className="sc-title operation-title"
                          >
                            demo1
                          </span>
                          <div
                            className="render-div"
                            id="demo1"
                            style={
                              Object {
                                "height": "calc(100% - 33px)",
                              }
                            }
                          />
                        </div>
                      </Shell>
                    </GridItem>
                    <GridItem
                      cancel=""
                      className=""
                      cols={2}
                      containerPadding={
                        Array [
                          20,
                          25,
                        ]
                      }
                      containerWidth={0}
                      h={9}
                      handle=""
                      i="demo2"
                      isDraggable={false}
                      isResizable={false}
                      key=".$demo2"
                      margin={
                        Array [
                          20,
                          25,
                        ]
                      }
                      maxH={Infinity}
                      maxRows={Infinity}
                      maxW={Infinity}
                      minH={1}
                      minW={1}
                      onDrag={[Function]}
                      onDragStart={[Function]}
                      onDragStop={[Function]}
                      onResize={[Function]}
                      onResizeStart={[Function]}
                      onResizeStop={[Function]}
                      rowHeight={5}
                      static={false}
                      useCSSTransforms={true}
                      usePercentages={false}
                      w={2}
                      x={0}
                      y={12}
                    >
                      <Shell
                        className="react-grid-item cssTransforms"
                        data-grid={
                          Object {
                            "detailpath": "http://www.baidu.com/",
                            "h": 9,
                            "i": "demo2",
                            "id": 118,
                            "imgurl": "http://47.95.1.229:9003/UploadFile/201808/dzx.png",
                            "maxw": 0,
                            "mealid": 89,
                            "minw": 0,
                            "path": "../../../../thirdModules/demo2/index.js",
                            "showdetail": true,
                            "still": false,
                            "style": Object {},
                            "title": "demo2",
                            "w": 3,
                            "x": 0,
                            "y": 12,
                          }
                        }
                        detailPath="http://www.baidu.com/"
                        key="demo2"
                        onEdit={[Function]}
                        showDelete={false}
                        showDetail={true}
                        showEdit={false}
                        showTitle={true}
                        size=""
                        style={
                          Object {
                            "MozTransform": "translate(20px,385px)",
                            "OTransform": "translate(20px,385px)",
                            "WebkitTransform": "translate(20px,385px)",
                            "height": "245px",
                            "msTransform": "translate(20px,385px)",
                            "position": "absolute",
                            "transform": "translate(20px,385px)",
                            "width": "-40px",
                          }
                        }
                        title="demo2"
                        type="component"
                      >
                        <div
                          className="Shell"
                          data-grid={
                            Object {
                              "detailpath": "http://www.baidu.com/",
                              "h": 9,
                              "i": "demo2",
                              "id": 118,
                              "imgurl": "http://47.95.1.229:9003/UploadFile/201808/dzx.png",
                              "maxw": 0,
                              "mealid": 89,
                              "minw": 0,
                              "path": "../../../../thirdModules/demo2/index.js",
                              "showdetail": true,
                              "still": false,
                              "style": Object {},
                              "title": "demo2",
                              "w": 3,
                              "x": 0,
                              "y": 12,
                            }
                          }
                          onClick={[Function]}
                          size=""
                          style={
                            Object {
                              "MozTransform": "translate(20px,385px)",
                              "OTransform": "translate(20px,385px)",
                              "WebkitTransform": "translate(20px,385px)",
                              "height": "245px",
                              "msTransform": "translate(20px,385px)",
                              "overflow": "auto",
                              "position": "absolute",
                              "transform": "translate(20px,385px)",
                              "width": "-40px",
                            }
                          }
                          type="component"
                        >
                          <span
                            onMouseDown={[Function]}
                            style={
                              Object {
                                "cursor": "pointer",
                                "fontSize": 17,
                                "position": "absolute",
                                "right": 10,
                                "top": 6,
                              }
                            }
                          >
                            详情 &gt;
                          </span>
                          <span
                            className="sc-title operation-title"
                          >
                            demo2
                          </span>
                          <div
                            className="render-div"
                            id="demo2"
                            style={
                              Object {
                                "height": "calc(100% - 33px)",
                              }
                            }
                          />
                        </div>
                      </Shell>
                    </GridItem>
                  </div>
                </ReactGridLayout>
              </ResponsiveReactGridLayout>
            </WidthProvider>
          </div>
        </Grid>
      </div>
    </div>
  </Navigation>
</Demo>
`;
